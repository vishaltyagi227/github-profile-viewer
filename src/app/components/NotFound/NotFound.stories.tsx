import React from 'react';
import { Story, Meta } from '@storybook/react';
import NotFound from '@components/NotFound';

export default {
  title: 'Components/NotFound',
  component: NotFound,
} as Meta;

const Template: Story = () => <NotFound />;

export const Basic = Template.bind({});
