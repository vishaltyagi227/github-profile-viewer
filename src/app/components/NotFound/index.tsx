import React from 'react';
import { Container, Typography } from '@mui/material';
import { Error } from '@mui/icons-material';

import { VARIANT } from '@constant/index';
import useStyles from './style';

const NotFound = () => {
  const classes = useStyles();

  return (
    <Container className={classes.notFoundContainer}>
      <div className={classes.notFoundWrapper}>
        <Typography variant={VARIANT.h3}>
          <Error className={classes.notFoundIcon} />
          404
        </Typography>
        <Typography variant={VARIANT.h3}>Not Found</Typography>
      </div>
    </Container>
  );
};

export default NotFound;
