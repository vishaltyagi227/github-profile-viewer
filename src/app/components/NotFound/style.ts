import { createStyles, makeStyles } from '@mui/styles';
import { Theme } from '@mui/material/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    notFoundContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      minHeight: '60vh',
      maxWidth: theme.breakpoints.values.xl,
    },
    notFoundWrapper: {
      width: '100%',
      textAlign: 'center',
    },
    notFoundIcon: {
      '&': theme.typography.h3,
      marginRight: theme.spacing(2),
    },
  })
);

export default useStyles;
