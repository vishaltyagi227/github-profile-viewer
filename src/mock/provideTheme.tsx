import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
