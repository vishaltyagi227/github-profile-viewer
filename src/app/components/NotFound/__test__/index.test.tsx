import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import NotFound from '../index';

describe('<NotFound />', () => {
  let container: RenderResult;

  afterEach(cleanup);

  it('Should render correctly', () => {
    container = render(provideTheme(<NotFound />));
    expect(
      container.getByRole('heading', { name: /404/i })
    ).toBeInTheDocument();
    expect(
      container.getByRole('heading', { name: /not found/i })
    ).toBeInTheDocument();
  });
});

describe('<NotFound /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with Snapshot', () => {
    const tree = renderer.create(provideTheme(<NotFound />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
