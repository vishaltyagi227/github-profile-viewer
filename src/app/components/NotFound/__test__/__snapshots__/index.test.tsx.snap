// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Snapshot Should match with Snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg makeStyles-notFoundContainer-4 css-1iin24e-MuiContainer-root"
>
  <div
    className="makeStyles-notFoundWrapper-5"
  >
    <h3
      className="MuiTypography-root MuiTypography-h3 css-q16rbx-MuiTypography-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-notFoundIcon-6 css-i4bv87-MuiSvgIcon-root"
        data-testid="ErrorIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      404
    </h3>
    <h3
      className="MuiTypography-root MuiTypography-h3 css-q16rbx-MuiTypography-root"
    >
      Not Found
    </h3>
  </div>
</div>
`;
